grover/grover.py

```diff
@@ -0,0 +1,98 @@
+from typing import List
+
+from app_oriented_benchmarks_adapted import grovers_benchmark
+
+import numpy as np
+
+import qiskit.compiler
+import qiskit
+
+from lsqecc.gates.gates_circuit import GatesCircuit
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.converters import circuit_to_dag
+from qiskit.transpiler import TransformationPass,PassManager
+
+from lsqecc.ls_instructions.ls_instructions_from_gates import LSInstructionsFromGatesGenerator
+
+
+class ForceReplaceWithDefinitionPass(TransformationPass):
+    def __init__(self, gate_names : List[str]):
+        super().__init__()
+        self.to_force_decompose = gate_names
+
+    def run(self, dag):
+        for node in dag.op_nodes():
+            if node.op.name in self.to_force_decompose:
+                dag.substitute_node_with_dag(node, circuit_to_dag(node.op.definition))
+        return dag
+
+class UnitaryToPhase(TransformationPass):
+    def run(self, dag):
+        for node in dag.op_nodes():
+            if node.op.name == "u1":
+                angle = node.op.params[0]
+                replacement = QuantumCircuit(1)
+                replacement.rz(angle, 0)
+                dag.substitute_node_with_dag(node, circuit_to_dag(replacement))
+            if node.op.name == "u2":
+                if node.op.params[0] in {0, -2*np.pi} and node.op.params[1] == np.pi:
+                    replacement = QuantumCircuit(1)
+                    replacement.h(0)
+                    dag.substitute_node_with_dag(node, circuit_to_dag(replacement))
+        return dag
```

Comment:
Do these passes make sense? `u2(0,pi)` is just a Hadamed gate right?

Reply:
As discussed on Zoom the math does check out, so `u2(0,pi)==h`. I don't understand what they are doing in the example

Reply:
I think thier formatting is wrong :sweat_smile: 

They have 3 separate examples. The spacing between them is broken.